test/sqllogictest/subquery.slt

```diff
@@ -175,10 +175,17 @@ query BBBBBBBBBBBBBBB
 ----
 true  false  false  true  false  false  NULL  true  NULL  NULL  NULL  NULL  NULL  NULL  NULL
 
-# TODO(benesch): this should yield a nullary result set, but presently
-# fails.
-# query error subquery has too few columns
-# SELECT 1 < ALL(SELECT * FROM nullary)
+query error subquery has 0 columns available but 1 columns specified
+SELECT 1 < ALL(SELECT * FROM nullary)
+
+query error Expected subselect to return 1 column, got 0 columns
+SELECT (SELECT);
+----
+
+query
+SELECT * FROM (SELECT);
```

Comment:
hmm pg on my laptop doesn't accept this query without an alias for the subquery

```
Test=# select * from (select);
ERROR:  subquery in FROM must have an alias
LINE 1: select * from (select);
                      ^
HINT:  For example, FROM (SELECT ...) [AS] foo.
Test=# select * from (select) as foo;
--
(1 row)
```

Reply:
i have zero context for whether thats a real issue or not in this case, just pointing it out

Reply:
Yup yup, that's a longstanding deviation between us and PostgreSQL. It'd be a really painful migration at this point. I filed it in #5273 just in case we want to tackle it though.